src/io/parquet/read/deserialize/simple.rs

```diff
@@ -347,130 +353,38 @@ fn timestamp_dict<'a, K: DictionaryKey, I: 'a + DataPages>(
     time_unit: TimeUnit,
 ) -> Result<ArrayIter<'a>> {
     if physical_type == &PhysicalType::Int96 {
-        if time_unit == TimeUnit::Nanosecond {
-            return Ok(dyn_iter(primitive::DictIter::<K, _, _, _, _>::new(
+        let (factor, is_multiplier) = unifiy_timestmap_unit(logical_type, time_unit);
+        return match (factor, is_multiplier) {
+            (a, true) => Ok(dyn_iter(primitive::DictIter::<K, _, _, _, _>::new(
                 pages,
                 DataType::Timestamp(TimeUnit::Nanosecond, None),
                 chunk_size,
-                int96_to_i64_ns,
-            )));
-        } else {
-            return Err(Error::nyi("Can't decode int96 to timestamp other than ns"));
-        }
+                move |x| int96_to_i64_ns(x) * a,
+            ))),
+            (a, false) => Ok(dyn_iter(primitive::DictIter::<K, _, _, _, _>::new(
+                pages,
+                DataType::Timestamp(TimeUnit::Nanosecond, None),
```

Comment:
@jorgecarleitao  Should we use `data_type` here? It may be incorrect to enforce the datatype to be `DataType::Timestamp(TimeUnit::Nanosecond, None)`.

Reply:
I agree.

I think that this comes from https://github.com/jorgecarleitao/arrow2/blob/main/src/io/parquet/read/schema/convert.rs#L181 where we infer int96 to be that, which is why here it works.